RabbitMQ.Stream.Client/RoutingClient.cs

```diff
@@ -80,15 +80,20 @@ string GetPropertyValue(IDictionary<string, string> connectionProperties, string
             var advertisedHost = GetPropertyValue(client.ConnectionProperties, "advertised_host");
             var advertisedPort = GetPropertyValue(client.ConnectionProperties, "advertised_port");
 
-            while (broker.Host != advertisedHost ||
-                   broker.Port != uint.Parse(advertisedPort))
+            var attempts = 0;
```

Comment:
Instead of a fixed value, we can use the `StreamInfo`.

So add a parameter to `LookupConnection(ClientParameters clientParameters, Broker broker)` something like: `LookupConnection(ClientParameters clientParameters, Broker broker, int attempts)` 


When used:
```csharp
public static async Task<IClient> LookupLeaderConnection(ClientParameters clientParameters,
            StreamInfo metaDataInfo)
        {
     
       var attempts =  1 + // this is the leader
            metaDataInfo.Replicas.Count // number of replicas 
    
        attempts = attempts *2; // just to be sure about loadbalancer redirect

       return await LookupConnection(clientParameters, metaDataInfo.Leader, attempts);
        }
```